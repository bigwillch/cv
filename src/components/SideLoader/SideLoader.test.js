import React from 'react';
import { SideLoader, SideLoaderNav } from 'Components';
import { SideLoaderContext } from 'Contexts';
import toJson from 'enzyme-to-json';

const sideLoaderContextValue = {
  state: {
    href: null,
    desc: null,
  },
  actions: {
    sideLoaded: jest.fn(() => 'loaded'),
    sideClose: jest.fn(() => 'close'),
  }
}

const createSideLoaderWrapper = (value = sideLoaderContextValue) => mount(
  <SideLoaderContext.Provider value={value}>
    <SideLoader />
  </SideLoaderContext.Provider>
)

describe('SideLoader', () => {

  it('should render empty if no href is set in context state', () => {
    const wrapper = createSideLoaderWrapper();
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.childAt(0).exists()).toEqual(false);
  });

  it('should render iframe and nav i,f href is set in context state', () => {
    const wrapper = createSideLoaderWrapper({
      ...sideLoaderContextValue,
      state: {
        href: 'examples/document',
        desc: 'Description',
      }
    });
    expect(toJson(wrapper)).toMatchSnapshot();

    expect(wrapper.find('iframe').prop('src')).toEqual('examples/document');
    expect(wrapper.find('iframe').prop('onLoad')).toEqual(sideLoaderContextValue.actions.sideLoaded);
    expect(wrapper.find('SideLoaderNav').prop('description')).toEqual('Description');
    expect(wrapper.find('SideLoaderNav').prop('sideClose')).toEqual(sideLoaderContextValue.actions.sideClose);
  });
});