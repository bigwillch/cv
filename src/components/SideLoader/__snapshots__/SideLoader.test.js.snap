// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideLoader should render empty if no href is set in context state 1`] = `<SideLoader />`;

exports[`SideLoader should render iframe and nav i,f href is set in context state 1`] = `
<SideLoader>
  <iframe
    onLoad={[MockFunction]}
    src="examples/document"
  />
  <SideLoaderNav
    description="Description"
    sideClose={[MockFunction]}
  >
    <div
      className=""
    >
      <nav>
        <div
          className="button button--back button--chunky"
          onClick={[MockFunction]}
          role="button"
          tabIndex="0"
        >
          Back
        </div>
        <div
          className="button button--up toggle"
          onClick={[Function]}
          role="button"
          tabIndex="0"
        />
      </nav>
      <p>
        Description
      </p>
    </div>
  </SideLoaderNav>
</SideLoader>
`;
